src/lib/dnssd/minimal_mdns/Server.cpp

```diff
@@ -230,229 +163,129 @@ CHIP_ERROR ServerBase::Listen(chip::Inet::EndPointManager<chip::Inet::UDPEndPoin
-            }
-
-            CHIP_ERROR err = CHIP_NO_ERROR;
-
-            /// The same packet needs to be sent over potentially multiple interfaces.
-            /// LWIP does not like having a pbuf sent over serparate interfaces, hence we create a copy
-            /// for sending via `CloneData`
-            ///
-            /// TODO: this wastes one copy of the data and that could be optimized away
-            chip::System::PacketBufferHandle tempBuf = data.CloneData();
-            if (tempBuf.IsNull())
-            {
-                // Not enough memory available to clone pbuf
-                err = CHIP_ERROR_NO_MEMORY;
-            }
-            else if (info->mAddressType == chip::Inet::IPAddressType::kIPv6)
-            {
-                err = udp->SendTo(mIpv6BroadcastAddress, port, std::move(tempBuf), udp->GetBoundInterface());
-            }
 #if INET_CONFIG_ENABLE_IPV4
-            else if (info->mAddressType == chip::Inet::IPAddressType::kIPv4)
-            {
-                err = udp->SendTo(mIpv4BroadcastAddress, port, std::move(tempBuf), udp->GetBoundInterface());
-            }
-#endif
-            else
-            {
-                // This is a general error of internal consistency: every address has a known type. Fail completely otherwise.
-                lastError = CHIP_ERROR_INCORRECT_STATE;
-                return chip::Loop::Break;
-            }
-
-            if (err == CHIP_NO_ERROR)
-            {
-                successes++;
-            }
-            else
-            {
-                failures++;
-                lastError = err;
-#if CHIP_DETAIL_LOGGING
-                char ifaceName[chip::Inet::InterfaceId::kMaxIfNameLength];
-                err = info->mInterfaceId.GetInterfaceName(ifaceName, sizeof(ifaceName));
-                if (err != CHIP_NO_ERROR)
-                    strcpy(ifaceName, "???");
-                ChipLogDetail(Discovery, "Warning: Attempt to mDNS broadcast failed on %s:  %s", ifaceName, lastError.AsString());
-#endif
-            }
-            return chip::Loop::Continue;
-        }))
+    if (addressType == chip::Inet::IPAddressType::kIPv4)
     {
-        return lastError;
+        VerifyOrReturnError(mIpv4Endpoint != nullptr, CHIP_ERROR_NOT_CONNECTED);
+        return mIpv4Endpoint->SendTo(mIpv4BroadcastAddress, port, std::move(data), interfaceId);
     }
-
-    if (failures != 0)
+    else if (addressType == chip::Inet::IPAddressType::kAny && (mIpv4Endpoint != nullptr))
     {
-        // if we had failures, log if the final status was success or failure, to make log reading
-        // easier. Some mDNS failures may be expected (e.g. for interfaces unavailable)
-        if (successes != 0)
+        // For "Any" we consider IPV4 optional, so only report errors
+        //
+        // The same packet needs to be sent over potentially multiple interfaces.
+        // LWIP does not like having a pbuf sent over serparate interfaces, hence we create a copy
```

Comment:
That's because it doesn't copy, and uses the pbuf provided, upping the refcount only, but using it as-is.